lib/active_merchant/billing/gateways/kushki.rb

```diff
@@ -0,0 +1,217 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class KushkiGateway < Gateway
+      self.display_name = "Kushki"
+      self.homepage_url = "https://www.kushkipagos.com"
+
+      self.test_url = "https://api-uat.kushkipagos.com/v1/"
+      self.live_url = "https://api.kushkipagos.com/v1/"
+
+      self.supported_countries = ["CO", "EC"]
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover, :diners_club]
+
+      def initialize(options={})
+        requires!(options, :public_merchant_id, :private_merchant_id)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        MultiResponse.run() do |r|
+          r.process { tokenize(amount, payment_method, options) }
+          r.process { charge(amount, r.authorization, options) }
+        end
+      end
+
+      def void(authorization, options={})
+        action = "void"
+
+        post = {}
+        add_invoice(action, post, nil, options)
+
+        commit(action, post, authorization)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Private-Merchant-Id: )\d+), '\1[FILTERED]').
+          gsub(%r((\"card\\\":{\\\"number\\\":\\\")\d+), '\1[FILTERED]').
+          gsub(%r((\"cvv\\\":\\\")\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      def tokenize(amount, payment_method, options)
+        action = "tokenize"
+
+        post = {}
+        add_invoice(action, post, amount, options)
+        add_payment_method(post, payment_method, options)
+
+        commit(action, post)
+      end
+
+      def charge(amount, authorization, options)
+        action = "charge"
+
+        post = {}
+        add_reference(post, authorization, options)
+        add_invoice(action, post, amount, options)
+
+        commit(action, post)
+      end
+
+      def add_invoice(action, post, money, options)
+        if action == "tokenize"
+          post[:totalAmount] = amount(money).to_f
```

Comment:
@jasonwebster The `amount()` method does that internally.

https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateway.rb#L254